dev/com.ibm.ws.jdbc/src/com/ibm/ws/rsadapter/impl/WSManagedConnectionFactoryImpl.java

```diff
@@ -853,13 +856,48 @@ private final Connection getConnectionUsingDriver(String userN, String password)
         try {
             conn = AccessController.doPrivileged(new PrivilegedExceptionAction<Connection>() {
                 public Connection run() throws Exception {
-                    Properties props = (Properties) dsConfig.get().vendorProps.clone(); //TODO would be nice if we could avoid cloning here
-                    String url = (String) props.remove("URL");
-                    if(user != null) {
-                        props.setProperty("user", user);
-                        props.setProperty("password", pwd);
+                    Hashtable<?, ?> vProps = dsConfig.get().vendorProps;
+                    Properties conProps = new Properties();
+                    String url = null;
+                    // convert property values to String and decode passwords 
+                    for (Map.Entry<?, ?> prop : vProps.entrySet()) {
+                        String name = (String) prop.getKey();
+                        Object value = prop.getValue();
+                        if (value instanceof String) {
+                            String str = (String) value;
+                            if ("URL".equals(name) || "url".equals(name)) {
+                                url = str;
+                                if (isTraceOn && tc.isDebugEnabled())
```

Comment:
I have concerns about tracing the url here (and anywhere else we may be tracing the properties) as there may be a password in the URL.